roles/ceph-osd/tasks/pool_names.yml

```diff
@@ -0,0 +1,144 @@
+---
+- assert:
+    that:
+      - ssd_pool is undefined
+      - hybrid_pool is undefined
+
+- name: gather existing cluster pool data if any
+  shell: rados lspools
+  register: existing_pools
+
+# if default pool doesn't exist
+- name: set pool names if default pool doesn't exist
+  set_fact:
+    ssd_pool: "rbd_ssd"
+    hybrid_pool: "rbd_hybrid"
+  when: '"default" not in existing_pools.stdout'
+
+# if default pool exist and ssd/hybrid pool doesn't exist
+- name: get first osd type
+  shell: ceph osd tree |sort -rn -k1 |grep -m1  -oP "(?<=host )\w+"
+  register: result_first_osd_host
+  when: ("default" in existing_pools.stdout) and
+        "rbd_ssd" not in existing_pools.stdout and
+        "rbd_hybrid" not in existing_pools.stdout
```

Comment:
This might look better if you make the when a list series, instead of a really long and string.

```
when:
  - existing_pools.stdout | search('default')
  - not existing_pools.stdout | search('rbd_ssd')
  - not existing_pools.stdout | search('rbd_hybrid')
```